src/zino/utils.py

```diff
@@ -25,3 +27,10 @@ def _parse_hexa_string_ip(ip: str) -> IPAddress:
 def _parse_colon_separated_ip(ip: str) -> IPAddress:
     """Parses IP addresses formatted with a colon symbol separating every octet, e.g. 7F:00:00:01"""
     return ip_address(bytes(OctetString(hexValue=ip.replace(":", ""))))
+
+
+def reverse_dns(ip: str) -> Optional[str]:
+    try:
+        return socket.gethostbyaddr(str(ip))[0]
+    except socket.herror:
+        return None
```

Comment:
This is a blocking socket operation and does not belong in an async event loop at all.

If we need to do async DNS requests, we need to look into other libraries, maybe something like `aiodnsresolver`.

Reply:
The most popularly used library I've been able to find in a few short minutes of research is https://github.com/saghul/aiodns

Reply:
derp, made some changes now